react/components/Notes.jsx

```diff
@@ -0,0 +1,189 @@
+import React, { useState, useEffect, useRef, useContext } from "react";
+import { Modal, ToastContext } from "vtex.styleguide";
+
+import styles from "./notes.css";
+
+const Notes = ({
+  wishlist: initialWishlist,
+  updateWishlist,
+  skuReference,
+  currentNotes,
+  productName,
+  productImage,
+  partNumber,
+  price,
+  currency,
+}) => {
+  const [show, setShow] = useState(false);
+  const [notes, setNotes] = useState("");
+
+  const wishlistRef = useRef(initialWishlist);
+
+  const { showToast } = useContext(ToastContext);
+
+  useEffect(() => {
+    if (currentNotes) {
+      setNotes(currentNotes);
+    }
+  }, [currentNotes]);
+
+  useEffect(() => {
+    wishlistRef.current = initialWishlist;
+  }, [initialWishlist]);
+
+  const handleNotesSubmit = async () => {
+    try {
+      const updatedProducts = wishlistRef.current.products.map((product) => {
+        if (product.skuCodeReference === skuReference) {
+          product.notes = notes;
+
+          return product;
+        }
+
+        return product;
+      });
+
+      await updateWishlist({
+        variables: {
+          wishlist: {
+            id: wishlistRef.current.id,
+            products: updatedProducts,
+          },
+        },
+      })
+        .then(() => {
```

Comment:
we shouldn't use .then with async await. 

```
await updateWishList(...)

showToast(...)

setShow(false)

```
Remove this try catch also

Reply:
Done